### ISSUE TITLE
Move images from community data to mediaserver

### ISSUE THREAD

**Issue by flexsurfer** [status-im/status-react#20284]
When we fetch community data there are lots of base64 images, and json is huge , we need to "move" all images to mediaserver

**Comment by ilmotta** [comment]
@qfrank, @flexsurfer as we recently discussed in issue https://github.com/status-im/status-mobile/issues/20285, moving encoded images out of the status-go community payload _may_ not bring much of an improvement to users (comment https://github.com/status-im/status-mobile/issues/20285#issuecomment-2153295981).

Have you guys checked removing the encoded images makes a difference, and by how much? We could quickly validate this idea by removing encoded images from the community payload returned by status-go. If no significant difference is measured for the end-user, then I think it's not worth the cost of development to migrate all images in the community payload to media servers now.

In general, I think most work on performance-related issues should first demonstrate the proposed solution does actually benefit users in a meaningful way. This can usually be quickly validated with a small sample and with dirty code. If the solution barely improves anything, then we abandon the original idea and iterate on another one, just like any other code optimization problem.

**Comment by qfrank** [comment]
> @qfrank, @flexsurfer as we recently discussed in issue #20285, moving encoded images out of the status-go community payload _may_ not bring much of an improvement to users (comment [#20285 (comment)](https://github.com/status-im/status-mobile/issues/20285#issuecomment-2153295981)).
> 
> Have you guys checked removing the encoded images makes a difference, and by how much? We could quickly validate this idea by removing encoded images from the community payload returned by status-go. If no significant difference is measured for the end-user, then I think it's not worth the cost of development to migrate all images in the community payload to media servers now.
> 
> In general, I think most work on performance-related issues should first demonstrate the proposed solution does actually benefit users in a meaningful way. This can usually be quickly validated with a small sample and with dirty code. If the solution barely improves anything, then we abandon the original idea and iterate on another one, just like any other code optimization problem.

I have asked @cammellos last week that:

> After this excellent discussion on this issue: https://github.com/status-im/status-mobile/issues/20285
> Now i'm not sure if I should continue work on this issue

and his answer was: "I think that's still valuable"

I agree as it could make the json look clearer.


> Have you guys checked removing the encoded images makes a difference, and by how much?

I didn't check yet.


WDYT? @ilmotta

**Comment by flexsurfer** [comment]
I would say that we should not consider this task as part of optimization for the community. We should view this task as a bug (incorrect implementation), a technical debt that affects the user. We should never use base64 for images; it is simply the wrong approach, impacting the performance of the entire application. We need to fix this wherever base64 is used. We should have a unified approach for handling images.

**Comment by cammellos** [comment]
Just for pros and cons:

Pros:

- We know from past experience and general understanding that base64 images are not performant as they are both slower to render as they use more resources (both cpu & memory). It was self evident in the message list, that's why we moved them to uri in the first place. They also add to the bridge, which we established already in the past is something that we need to be careful with.
- We have an established method (uri) that should be followed everywhere, base64 should be the exception.

Cons:

- In this case, it's more a matter than code hygiene, it's true that we haven't quantified performance benefits and we are acting more on principles rather than specifics (proactive vs reactive).

I don't have a strong opinion on the matter of whether we should fix the issue (as it should not have been coded this way in the first place), but we should be certainly avoiding this in the future as we know it can be problematic, happy to go with your call @ilmotta either ways

(if anyone has more to add please feel free to add pros/cons, those are the ones that came to my mind first)

**Comment by ilmotta** [comment]
I agree with the principles you outlined @cammellos and also with the perspective of a tech-debt you shared @flexsurfer.

On a theoretical ground there are a multitude of things we could do, but not all are worth our time. The example where we removed Jump To is a great one, that showed that in a matter of hours Andrea could check a performance improvement from a quick experiment done by @Parveshdhull to disable it. What I find good about this example is that we first checked the results of an experiment and only then committed to change for real. This mindset I think is paramount.

From a prioritization perspective, one question is: _What is or what are the most impactful things we can do to optimize the app to users?_ We focus on those and only start to attack small fish after we think we ran out of ideas. Otherwise it feels a bit like premature optimization.

Of course there's a balance to manage tech-debt. From that perspective I think the task is fine as Andrey explained. Just from the performance standpoint, which this issue is linked to https://github.com/status-im/status-mobile/issues/20283 that I find harder to understand why we'd do this instead of something probably more impactful, like refactoring how we use community subscriptions.
